src/Extensions/Diagnostics.HealthChecks/Internal/HttpHealthCheckParameters.cs

```diff
@@ -12,56 +11,26 @@ namespace Microsoft.Omex.Extensions.Diagnostics.HealthChecks
 {
 	internal class HttpHealthCheckParameters : HealthCheckParameters
 	{
-		private static IReadOnlyDictionary<string, IEnumerable<string>> s_emptyHeaders = new ReadOnlyDictionary<string, IEnumerable<string>>(new Dictionary<string, IEnumerable<string>>(0));
-
-		public string EndpointName { get; }
-
-		public Uri RelativeUri { get; }
-
-		public HttpMethod Method { get; }
-
 		public HttpStatusCode ExpectedStatus { get; }
 
-		public string Scheme { get; }
-
-		public IReadOnlyDictionary<string, IEnumerable<string>> Headers { get; }
+		public HttpRequestMessage RequestMessage { get; }
 
 		public Func<HttpResponseMessage, HealthCheckResult, HealthCheckResult>? AdditionalCheck { get; }
 
 		/// <summary>
 		/// Creates HttpHealthCheckParameters instance
 		/// </summary>
 		/// <remarks>
-		///	Parameter description provided in a public method for creation of http health check <see cref="HealthChecksBuilderExtensions.AddHttpEndpointCheck"/>
+		/// Parameter description provided in a public method for creation of http health check <see cref="HealthChecksBuilderExtensions"/>
 		/// </remarks>
 		public HttpHealthCheckParameters(
-			string endpointName,
-			Uri relatedUri,
-			HttpMethod? method,
-			string? scheme,
-			IReadOnlyDictionary<string, IEnumerable<string>>? headers,
+			HttpRequestMessage httpRequestMessage,
 			HttpStatusCode? expectedStatus,
 			Func<HttpResponseMessage, HealthCheckResult, HealthCheckResult>? additionalCheck,
 			KeyValuePair<string, object>[] reportData)
 				: base(reportData)
 		{
-			EndpointName = string.IsNullOrWhiteSpace(endpointName)
-				? throw new ArgumentException("Invalid endpoint name", nameof(endpointName))
-				: endpointName;
-
-			RelativeUri = relatedUri.IsAbsoluteUri
-				? throw new ArgumentException("Absolute uri not allowed", nameof(relatedUri))
-				: relatedUri;
-
-			Method = method ?? HttpMethod.Get;
-
-			Scheme = scheme == null
-				? Uri.UriSchemeHttp
-				: Uri.CheckSchemeName(scheme)
-					? scheme
-					: throw new ArgumentException("Invalid uri scheme", nameof(scheme));
-
-			Headers = headers ?? s_emptyHeaders;
+			RequestMessage = httpRequestMessage == null ? throw new ArgumentNullException(nameof(httpRequestMessage)) : httpRequestMessage;
```

Comment:
This class is internal so the only usage would be in this project with nullability enabled, so I think just having a non-nullable type is enough. But if you insist then shorter form could be used like `RequestMessage = httpRequestMessage ?? throw new ArgumentNullException(nameof(httpRequestMessage));`.